libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -366,8 +408,8 @@ void AC_PosControl::set_alt_target_from_climb_rate_ff(float climb_rate_cms, floa
 {
     // calculated increased maximum acceleration if over speed
     float accel_z_cms = _accel_z_cms;
-    if (_vel_desired.z < _speed_down_cms && !is_zero(_speed_down_cms)) {
-        accel_z_cms *= POSCONTROL_OVERSPEED_GAIN_Z * _vel_desired.z / _speed_down_cms;
+    if (_vel_desired.z < -fabsf(_speed_down_cms) && !is_zero(_speed_down_cms)) {
```

Comment:
```suggestion
    if (_vel_desired.z < -_speed_down_cms && is_positive(_speed_down_cms)) {
```

Reply:
This is used because the sign has been reversed on occasion resulting in nasty results. I too would prefer that it was guaranteed to be expressed as a positive number. Human nature and all....

Reply:
the use of _speed_down_cms is quite tricky because it should always be a negative number.  This makes it easy to make mistakes and there was actually a serious bug on this line that has now been fixed by reverting it back to what it looks like in master.